src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,380 @@
+        require(gemDecimals <= daiDecimals, "RwaOutputConduit3/invalid-gem-decimals");
+        to18ConvertionFactor = 10**(daiDecimals - gemDecimals);
+
+        // Give unlimited approve to PSM
+        dai.approve(_psm, type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "RwaOutputConduit3/not-authorized");
+        _;
+    }
+
+    modifier isMate() {
+        require(may[msg.sender] == 1, "RwaOutputConduit3/not-mate");
+        _;
+    }
+
+    /*//////////////////////////////////
+               Authorization
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /**
+     * @notice Grants `usr` operator access to this contract.
+     * @param usr The user address.
+     */
+    function hope(address usr) external auth {
+        can[usr] = 1;
+        emit Hope(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` operator access from this contract.
+     * @param usr The user address.
+     */
+    function nope(address usr) external auth {
+        can[usr] = 0;
+        emit Nope(usr);
+    }
+
+    /**
+     * @notice Whitelist `who` address for `pick`.
+     * @param who The user address.
+     */
+    function kiss(address who) external auth {
+        bud[who] = 1;
+        emit Kiss(who);
+    }
+
+    /**
+     * @notice Remove `who` address from `pick` whitelist.
+     * @param who The user address.
+     */
+    function diss(address who) external auth {
+        if (to == who) to = address(0);
+        bud[who] = 0;
+        emit Diss(who);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"quitTo"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            require(data != address(0), "RwaOutputConduit3/invalid-quit-to-address");
```

Comment:
we usually try to avoid validations in the file function (and do the checks in the spell). Here I think it's even safer to remove as you check for address(0) in `_doQuit`

Reply:
done